### PR TITLE
PerlDDS additions to detect unresponsive VMs

### PR DESCRIPTION
**Pull request by jonesc-oci**

### CODE REVIEW

**Comment by jonesc-oci** [comment]
Here's the smoke test running with a responsiveness monitor that logs console message if it detects more than 200ms of clock drift.

```
$ perl run_test.pl 2>&1 | tee log
Response monitor running with 1 second delay and 200 msecs of drift
/home/chip/OpenDDS/tests/DCPS/RtpsRelay/Smoke/monitor -DCPSConfigFile monitor.ini -DCPSDebugLevel 1 -ORBVerboseLogging 1 -DCPSTransportDebugLevel 1 -ORBLogFile monitor.log -DCPSPendingTimeout 3 
...
...
relay2 PID: 44130 started at 2022-03-31 15:17:16
Response monitor reports 252.471208572388 milliseconds drift.
Response monitor reports 201.620817184448 milliseconds drift.
/home/chip/OpenDDS/tests/DCPS/RtpsRelay/Smoke/publisher -ORBDebugLevel 1 -DCPSConfigFile pub_rtps.ini -DCPSDebugLevel 1 -ORBVerboseLogging 1 -DCPSTransportDebugLevel 1 -ORBLogFile publisher.log -DCPSPendingTimeout 3
...
...
2022-03-31 15:17:37.424@LM_DEBUG@(44156|44156) Service_Participant::~Service_Participant

<<<<<  end subscriber.log  >>>>>

test FAILED.
Response monitor pid 44127 stopped.
```

File: bin/PerlDDS/Response_Monitor.pm [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+            child => $pid }, $self;
+  } else {
+    print "Response monitor running with $delay second delay and $drift msecs of drift\n";
+
+    # child
+    my $last;
+    my $now;
+    my $diff_ms;
+
+    while(1) {
```
Comment (iguessthislldo) [inline bin/PerlDDS/Response_Monitor.pm:28]:
```suggestion
    while (1) {
```

File: bin/PerlDDS/Response_Monitor.pm [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+      $last = Time::HiRes::gettimeofday();
+      Time::HiRes::sleep $delay;
+      $now = Time::HiRes::gettimeofday();
+      $diff_ms = 1000 * ($now - $last - $delay);
+
+      if ( $diff_ms > $drift) {
```
Comment (iguessthislldo) [inline bin/PerlDDS/Response_Monitor.pm:34]:
```suggestion
      if ($diff_ms > $drift) {
```

File: bin/PerlDDS/Response_Monitor.pm [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+        print "Response monitor reports $diff_ms milliseconds drift.\n";
```
Comment (iguessthislldo) [inline bin/PerlDDS/Response_Monitor.pm:35]: Do we want this to be a warning so autobuild picks up on this?

**Review by iguessthislldo — CHANGES_REQUESTED** [review]

File: bin/PerlDDS/Response_Monitor.pm (hunk 3 above)
Comment (jonesc-oci) [reply bin/PerlDDS/Response_Monitor.pm:35]: Good idea. Does autobuild just look for "WARNING:" in the message?

Comment (iguessthislldo) [reply bin/PerlDDS/Response_Monitor.pm:35]: Yes, I'm not sure what it's exactly looking for, but that should do it.

**Pull request merged**